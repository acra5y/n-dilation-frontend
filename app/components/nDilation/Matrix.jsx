import React from "react";
import styled from "styled-components";

const Matrix = styled.div`
    overflow: scroll;
    display: grid;
    grid-gap: 4px;
    grid-template-columns: ${({ dimension }) => `repeat(${dimension}, 1fr);`};
`;
const WrapperWithPaddingOnOverflow = styled.div`
    padding: 0.25rem 0.5rem;
    border-left: 1px solid black;
    border-right: 1px solid black;
    max-width: 95vw;
`;

export default ({ matrixInRowMajorOrder }) => (
    <WrapperWithPaddingOnOverflow>
        <Matrix dimension={Math.round(Math.sqrt(matrixInRowMajorOrder.length))}>
            {matrixInRowMajorOrder.map((entry, i) => (
                <span key={`entry-${i}`}>{entry}</span>
            ))}
        </Matrix>
    </WrapperWithPaddingOnOverflow>
);
