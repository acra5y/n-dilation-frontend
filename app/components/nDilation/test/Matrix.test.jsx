import React from "react";
import { shallow } from "enzyme";

import Matrix from "../Matrix";

const defaultProps = {
    matrixInRowMajorOrder: [0, 1, 2, 3, 4, 5, 6, 7, 8],
};

const render = overrideProps =>
    shallow(<Matrix {...defaultProps} {...overrideProps} />);

describe("Matrix", () => {
    it("should render a grid", () => {
        const component = render();

        expect(component).toMatchSnapshot();
    });
});
