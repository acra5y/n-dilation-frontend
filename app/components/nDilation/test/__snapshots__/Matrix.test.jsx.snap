// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matrix should render a grid 1`] = `
<Matrix__WrapperWithPaddingOnOverflow>
  <Matrix
    dimension={3}
  >
    <span
      key="entry-0"
    >
      0
    </span>
    <span
      key="entry-1"
    >
      1
    </span>
    <span
      key="entry-2"
    >
      2
    </span>
    <span
      key="entry-3"
    >
      3
    </span>
    <span
      key="entry-4"
    >
      4
    </span>
    <span
      key="entry-5"
    >
      5
    </span>
    <span
      key="entry-6"
    >
      6
    </span>
    <span
      key="entry-7"
    >
      7
    </span>
    <span
      key="entry-8"
    >
      8
    </span>
  </Matrix>
</Matrix__WrapperWithPaddingOnOverflow>
`;
